channeldb/channel.go

```diff
@@ -1940,12 +1940,12 @@ func deserializeLogUpdates(r io.Reader) ([]LogUpdate, error) {
 	return logUpdates, nil
 }
 
-func serializeCommitDiff(w io.Writer, diff *CommitDiff) error {
+func serializeCommitDiff(w io.Writer, diff *CommitDiff) error { // nolint: dupl
 	if err := serializeChanCommit(w, &diff.Commitment); err != nil {
 		return err
 	}
 
-	if err := diff.CommitSig.Encode(w, 0); err != nil {
+	if err := WriteElements(w, diff.CommitSig); err != nil {
```

Comment:
For the other reviewer, this is where the encode/decode change actually happens: we'll now use `WriteElements` which'll use the proper `lnwire` message encoding (including the type and length prefix) instead of `CommitSig.Encode` which writes out the raw bytes in-place.